src/pytest_adf/pytest_adf.py

```diff
@@ -161,3 +208,162 @@ def _poll_adf_until(adf_client, az_resource_group, adf_name, pipeline_run_id,
+    Warning: if target deleting file is not there, BlobNotFound exception will be raised.
+    
+    Args:
+        blob_service_client (BlobServiceClient): Azure Storage Account service client
+        trigger_file_path (str): Path of trigger file in storage account container
+        container_name (str): Name of target container to upload the trigger file.
+
+    Returns:
+        blob_properties, including last modified time
+    """
+    container_client = blob_service_client.get_container_client(container_name)
+
+    # Delete target event trigger file from storage account
+    container_client.delete_blob(trigger_file_path)
+    deleted_time = datetime.utcnow()
+    blob_properties = {
+        "etag": "",
+        "last_modified": deleted_time
+    }
+    
+    return blob_properties
+
+
+def _get_triggered_pipeline_run_id(adf_client, az_resource_group, adf_name, pipeline_name, trigger_name, blob_event_type, trigger_file_last_modified_time, trigger_file_etag, poll_interval=5):
+    """Helper functioin to get triggered pipeline run id by looking for expected trigger-run with last moified time and eTag of target event trigger file.
+    Warning! TimeoutError exception may be raised if it still fails to find the expected trigger-run after the configured max time.
+    """
+    max_iterations = int(FETCHING_TRIGGER_RUNS_MAX_TIME_IN_SECONDS / poll_interval)
+    iteration = 0
+    found_trigger_run = False
+    triggered_pipeline_run_id = ""
+    while (found_trigger_run == False):
+        LOG.info("Polling trigger-run by trigger name {}, last modified time and eTag of target trigger file.".format(trigger_name))
+        
+        if not found_trigger_run:
+            run_query_filter = RunQueryFilter(operand=RunQueryFilterOperand.TRIGGER_NAME, operator=RunQueryFilterOperator.EQUALS, values=[trigger_name])
+            run_query_order_by = RunQueryOrderBy(order_by=RunQueryOrderByField.TRIGGER_RUN_TIMESTAMP, order=RunQueryOrder.DESC)
+            filter_params = RunFilterParameters(last_updated_after=trigger_file_last_modified_time, 
+                                                last_updated_before=trigger_file_last_modified_time + timedelta(minutes = FETCHING_TRIGGER_RUNS_TIME_RANGE_IN_MINUTES),
+                                                filters=[run_query_filter],
+                                                order_by=[run_query_order_by])
+            
+            trigger_runs_query_response = adf_client.trigger_runs.query_by_factory(az_resource_group, 
+                                                                                   adf_name, 
+                                                                                   filter_params)
+            trigger_runs = trigger_runs_query_response.value
+            if (len(trigger_runs) > 0):
+                for trigger_run in trigger_runs:
+                    event_payload = trigger_run.properties.get("EventPayload")
```

Comment:
Perhaps add another `LOG.INFO` to log the resulting pipelines runs found.

Also, would be good to add an `else` block then add `LOG.INFO` that no pipelines have been found.